src/lib/concat_path_file.c

```diff
@@ -57,3 +57,28 @@ char *concat_path_basename(const char *path, const char *filename)
     free(abspath);
     return name;
 }
+
+bool str_is_correct_filename(const char *str)
+{
+#define NOT_PRINTABLE(c) (c < ' ' || c == 0x7f)
+
+    if (NOT_PRINTABLE(*str) || *str == '/' || *str == '\0')
+        return false;
+    ++str;
+
+    if (NOT_PRINTABLE(*str) || *str =='/' || (*str == '\0' && *(str-1) == '.'))
+        return false;
+    ++str;
+
+    if (NOT_PRINTABLE(*str) || *str =='/' || (*str == '\0' && *(str-1) == '.' && *(str-2) == '.'))
+        return false;
+    ++str;
+
+    for (unsigned i = 0; *str != '\0' && i < 61; ++str, ++i)
```

Comment:
This code need more comments, all the security patches I've developed in the last two weeks need documentation.

Ad 61:
This function accepts only strings 64 characters long. 61 + 3 above. 

Ad the for loop:
0. you are right but I'm not going to rewrite it because:
1. this function needs to be fast
2. check for '.' make sense only for the first two characters
3. searching for '.' in the loop would be redundant because